src/Makefile

```diff
@@ -116,7 +116,7 @@ ifeq ($(feature-disassembler-init-styled), 1)
     CFLAGS += -DDISASM_INIT_STYLED
 endif
 
-LIBS = $(LIBBPF) -lelf -lz
+LIBS = $(LIBBPF) -lelf -lz $(EXTRA_LIBS)
```

Comment:
This file is kept in sync, as much with possible, with bpftool's Makefile from kernel, so I'd rather avoid changing it in this repo if there's another way to do it. Why is the flag required here? Shouldn't feature detection in the build system figure out what libraries are required?

Reply:
The feature detection does not work in the container. It seems that `src/Makefile.feature` detects the following:
- `feature-libbfd` =1
- `feature-libbfd-liberty` not defined
- `feature-libbfd-liberty-z` not defined

The detection program in `src/Makefile.feature` just uses `bfd_demangle` but it does not use other symbols, see errors below. I am not sure how it should be changed.

```
$ docker build -f Dockerfile -t bpftool .
...
  LINK     bpftool
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(plugin.o): in function `try_load_plugin':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/plugin.c:354: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elf64-x86-64.o): in function `elf_x86_64_output_arch_local_syms':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf64-x86-64.c:4821: undefined reference to `htab_traverse'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elfxx-x86.o): in function `elf_x86_link_hash_table_free':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elfxx-x86.c:700: undefined reference to `htab_delete'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elfxx-x86.c:702: undefined reference to `objalloc_free'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elfxx-x86.o): in function `_bfd_elf_x86_get_local_sym_hash':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elfxx-x86.c:588: undefined reference to `htab_find_slot_with_hash'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elfxx-x86.c:601: undefined reference to `_objalloc_alloc'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elfxx-x86.o): in function `_bfd_x86_elf_link_hash_table_create':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elfxx-x86.c:790: undefined reference to `htab_try_create'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elfxx-x86.c:794: undefined reference to `objalloc_create'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elfxx-x86.o): in function `elf_x86_link_hash_table_free':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elfxx-x86.c:700: undefined reference to `htab_delete'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elfxx-x86.c:702: undefined reference to `objalloc_free'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elfxx-x86.o): in function `_bfd_x86_elf_size_dynamic_sections':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elfxx-x86.c:2171: undefined reference to `htab_traverse'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elflink.o): in function `elf_link_add_object_symbols':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elflink.c:5595: undefined reference to `objalloc_free_block'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elflink.c:5844: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elflink.o): in function `bfd_elf_size_dynamic_sections':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elflink.c:7075: undefined reference to `lbasename'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elflink.c:6843: undefined reference to `lbasename'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elflink.o): in function `elf_link_input_bfd':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elflink.c:11162: undefined reference to `lbasename'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(pei-x86_64.o): in function `pex64_bfd_print_pdata_section':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/pei-x86_64.c:622: undefined reference to `xmalloc'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/pei-x86_64.c:698: undefined reference to `xmalloc'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libopcodes.a(i386-dis.o): in function `putop':
/build/binutils-spemQQ/binutils-2.38/builddir-single/opcodes/../../opcodes/i386-dis.c:10229: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archures.o): in function `bfd_default_scan':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archures.c:1208: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(bfd.o): in function `_bfd_doprnt_scan':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/bfd.c:1167: undefined reference to `_sch_istable'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/bfd.c:1183: undefined reference to `_sch_istable'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/bfd.c:1220: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(bfd.o):/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/bfd.c:1195: more undefined references to `_sch_istable' follow
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(bfd.o): in function `bfd_demangle':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/bfd.c:2428: undefined reference to `cplus_demangle'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/bfd.c:2428: undefined reference to `cplus_demangle'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/bfd.c:2428: undefined reference to `cplus_demangle'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(bfd.o): in function `bfd_errmsg':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/bfd.c:812: undefined reference to `xstrerror'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(hash.o): in function `bfd_hash_table_free':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/hash.c:426: undefined reference to `objalloc_free'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(hash.o): in function `bfd_hash_table_init_n':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/hash.c:385: undefined reference to `objalloc_create'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/hash.c:392: undefined reference to `_objalloc_alloc'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(hash.o): in function `bfd_hash_insert':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/hash.c:536: undefined reference to `_objalloc_alloc'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(hash.o): in function `bfd_hash_lookup':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/hash.c:486: undefined reference to `_objalloc_alloc'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(hash.o): in function `bfd_hash_allocate':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/hash.c:623: undefined reference to `_objalloc_alloc'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(opncls.o): in function `find_separate_debug_file':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:1507: undefined reference to `lrealpath'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(opncls.o): in function `_bfd_new_bfd':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:74: undefined reference to `objalloc_create'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:74: undefined reference to `objalloc_create'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:87: undefined reference to `objalloc_free'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(opncls.o): in function `_bfd_free_cached_info':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:169: undefined reference to `objalloc_free'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:169: undefined reference to `objalloc_free'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(opncls.o): in function `_bfd_delete_bfd':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:129: undefined reference to `objalloc_free'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(opncls.o): in function `bfd_alloc':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:1039: undefined reference to `_objalloc_alloc'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(opncls.o): in function `bfd_create_gnu_debuglink_section':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:1715: undefined reference to `lbasename'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(opncls.o): in function `bfd_fill_in_gnu_debuglink_section':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:1809: undefined reference to `lbasename'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(opncls.o): in function `_bfd_delete_bfd':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:129: undefined reference to `objalloc_free'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:129: undefined reference to `objalloc_free'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:129: undefined reference to `objalloc_free'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:129: undefined reference to `objalloc_free'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:129: undefined reference to `objalloc_free'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(opncls.o):/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:129: more undefined references to `objalloc_free' follow
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(opncls.o): in function `bfd_release':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/opncls.c:1076: undefined reference to `objalloc_free_block'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(binary.o): in function `mangle_name':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/binary.c:139: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(ihex.o): in function `ihex_bad_byte':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/ihex.c:221: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(ihex.o): in function `ihex_read_section':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/ihex.c:574: undefined reference to `_hex_value'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(ihex.o): in function `ihex_object_p':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/ihex.c:513: undefined reference to `_hex_value'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(ihex.o): in function `ihex_init':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/ihex.c:168: undefined reference to `hex_init'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(srec.o): in function `srec_bad_byte':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/srec.c:251: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(srec.o): in function `srec_init':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/srec.c:186: undefined reference to `hex_init'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(srec.o): in function `srec_read_section':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/srec.c:742: undefined reference to `_hex_value'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(srec.o): in function `srec_scan':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/srec.c:469: undefined reference to `_hex_value'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/srec.c:373: undefined reference to `_sch_istable'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/srec.c:425: undefined reference to `_hex_value'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(srec.o): in function `srec_init':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/srec.c:186: undefined reference to `hex_init'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(srec.o): in function `srec_object_p':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/srec.c:654: undefined reference to `_hex_value'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(srec.o): in function `srec_init':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/srec.c:186: undefined reference to `hex_init'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(tekhex.o): in function `getvalue':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/tekhex.c:279: undefined reference to `_hex_value'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(tekhex.o): in function `getsym':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/tekhex.c:304: undefined reference to `_hex_value'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(tekhex.o): in function `first_phase':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/tekhex.c:379: undefined reference to `_hex_value'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(tekhex.o): in function `tekhex_init':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/tekhex.c:213: undefined reference to `hex_init'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(tekhex.o): in function `pass_over':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/tekhex.c:540: undefined reference to `_hex_value'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(tekhex.o): in function `tekhex_object_p':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/tekhex.c:614: undefined reference to `_hex_value'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(verilog.o): in function `verilog_init':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/verilog.c:342: undefined reference to `hex_init'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elf.o): in function `_bfd_elf_is_local_label_name':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf.c:9205: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elf-eh-frame.o): in function `cie_compute_hash':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf-eh-frame.c:261: undefined reference to `iterative_hash'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf-eh-frame.c:262: undefined reference to `iterative_hash'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf-eh-frame.c:263: undefined reference to `iterative_hash'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf-eh-frame.c:264: undefined reference to `iterative_hash'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf-eh-frame.c:265: undefined reference to `iterative_hash'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elf-eh-frame.o):/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf-eh-frame.c:266: more undefined references to `iterative_hash' follow
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elf-eh-frame.o): in function `find_merged_cie':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf-eh-frame.c:1300: undefined reference to `htab_find_slot_with_hash'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf-eh-frame.c:1296: undefined reference to `htab_try_create'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elf-eh-frame.o): in function `_bfd_elf_discard_section_eh_frame_hdr':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf-eh-frame.c:1626: undefined reference to `htab_delete'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(dwarf2.o): in function `hash_abbrev':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/dwarf2.c:989: undefined reference to `htab_hash_pointer'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(dwarf2.o): in function `read_abbrevs':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/dwarf2.c:1042: undefined reference to `htab_find_slot'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(dwarf2.o): in function `_bfd_dwarf2_find_symbol_bias':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/dwarf2.c:4970: undefined reference to `xcalloc'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/dwarf2.c:4970: undefined reference to `htab_create_alloc'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/dwarf2.c:4978: undefined reference to `htab_find_slot'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/dwarf2.c:4997: undefined reference to `htab_find'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/dwarf2.c:5008: undefined reference to `htab_delete'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(dwarf2.o): in function `_bfd_dwarf2_cleanup_debug_info':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/dwarf2.c:5363: undefined reference to `htab_delete'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(dwarf2.o): in function `_bfd_dwarf2_slurp_debug_info':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/dwarf2.c:4740: undefined reference to `htab_create_alloc'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/dwarf2.c:4745: undefined reference to `htab_create_alloc'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(dwarf2.o): in function `hash_asymbol':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/dwarf2.c:4938: undefined reference to `htab_hash_string'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(elf32-i386.o): in function `elf_i386_output_arch_local_syms':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/elf32-i386.c:4172: undefined reference to `htab_traverse'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(cofflink.o): in function `coff_link_add_symbols':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/cofflink.c:551: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(plugin.o): in function `build_plugin_list':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/plugin.c:510: undefined reference to `make_relative_prefix'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/plugin.c:533: undefined reference to `concat'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `adjust_relative_path':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1426: undefined reference to `getpwd'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1430: undefined reference to `lrealpath'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1433: undefined reference to `lrealpath'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1447: undefined reference to `filename_ncmp'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `_bfd_look_for_bfd_in_cache':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:311: undefined reference to `htab_find'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `_bfd_add_bfd_to_archive_cache':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:372: undefined reference to `htab_find_slot'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:361: undefined reference to `htab_create_alloc'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `_bfd_generic_read_ar_hdr_mag':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:538: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `_bfd_append_relative_path':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:632: undefined reference to `lbasename'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `_bfd_look_for_bfd_in_cache':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:311: undefined reference to `htab_find'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `find_nested_archive':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:406: undefined reference to `filename_cmp'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:416: undefined reference to `filename_cmp'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `_bfd_look_for_bfd_in_cache':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:311: undefined reference to `htab_find'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `_bfd_construct_extended_name_table':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1597: undefined reference to `filename_cmp'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `normalize':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1387: undefined reference to `lbasename'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `_bfd_construct_extended_name_table':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1643: undefined reference to `filename_ncmp'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1691: undefined reference to `filename_cmp'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1643: undefined reference to `filename_ncmp'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `normalize':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1387: undefined reference to `lbasename'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1387: undefined reference to `lbasename'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `_bfd_bsd44_write_ar_hdr':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1806: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `normalize':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1387: undefined reference to `lbasename'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `bfd_bsd_truncate_arname':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:2055: undefined reference to `lbasename'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `normalize':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:1387: undefined reference to `lbasename'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `bfd_gnu_truncate_arname':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:2087: undefined reference to `lbasename'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `_bfd_unlink_from_archive_parent':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:2837: undefined reference to `htab_find_slot'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:2841: undefined reference to `htab_clear_slot'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:2841: undefined reference to `htab_clear_slot'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(archive.o): in function `_bfd_archive_close_and_cleanup':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:2866: undefined reference to `htab_traverse_noresize'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/archive.c:2867: undefined reference to `htab_delete'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(cache.o): in function `bfd_open_file':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/cache.c:633: undefined reference to `unlink_if_ordinary'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(compress.o): in function `bfd_is_section_compressed_with_header':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/compress.c:461: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(compress.o): in function `bfd_compress_section_contents':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/compress.c:131: undefined reference to `compressBound'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/compress.c:178: undefined reference to `compress'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(corefile.o): in function `generic_core_file_matches_executable_p':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/corefile.c:188: undefined reference to `filename_cmp'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(stabs.o): in function `_bfd_link_section_stabs':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/stabs.c:352: undefined reference to `_sch_istable'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(syms.o): in function `bfd_decode_symclass':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/syms.c:712: undefined reference to `_sch_toupper'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libbfd.a(syms.o): in function `_bfd_stab_section_find_nearest_line':
/build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/syms.c:1413: undefined reference to `filename_ncmp'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/bfd/../../bfd/syms.c:1414: undefined reference to `filename_cmp'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/11/../../../x86_64-linux-gnu/libopcodes.a(disassemble.o): in function `remove_whitespace_and_extra_commas':
/build/binutils-spemQQ/binutils-2.38/builddir-single/opcodes/../../opcodes/disassemble.c:804: undefined reference to `_sch_istable'
/usr/bin/ld: /build/binutils-spemQQ/binutils-2.38/builddir-single/opcodes/../../opcodes/disassemble.c:812: undefined reference to `_sch_istable'
collect2: error: ld returned 1 exit status
make: *** [Makefile:209: bpftool] Error 1
The command '/bin/sh -c cd /src/src ; 	export CFLAGS="--static -DPACKAGE_VERSION=1" ; 	make clean ; 	make -j 6' returned a non-zero code: 2
```